omegaconf/omegaconf.py

```diff
@@ -77,7 +76,7 @@
 # -  in OmegaConf.create() to differentiate between creating an empty {} DictConfig
 #    and creating a DictConfig with None content
 # - in env() to detect between no default value vs a default value set to None
-_EMPTY_MARKER_ = object()
+_EMPTY_MARKER_ = "_OMEGACONF_EMPTY_MARKER_"
```

Comment:
Yeah, I wasn't sure if you'd care about someone using the string `"_OMEGACONF_EMPTY_MARKER_"` in their own code. I had in mind something a bit different though regarding a potential fix:
```python
class StringMarker(str):
    ...
```
then
```python
_EMPTY_MARKER_ = StringMarker()
```
Motivation:
* I don't really see the point of having these markers hold data (their name should be enough)
* Inheriting from `str` avoids typing problems with `Optional[str] = _EMPTY_MARKER_` (it also means we may need other markers for other types). The alternative is to have a single `Marker` class and use `Union[Optional[str], Marker]` instead, but it makes things a bit ugly IMO.

Reply:
> I don't really see the point of having these markers hold data (their name should be enough)

Makes it easier to debug. (The string as opposed to some unknown object).

> Inheriting from str avoids typing problems with Optional[str] = _EMPTY_MARKER_

But does not work for any other type.
for example, `dict.pop(key, default)` should work for any default type. 

I think a marker should not be typed, it's just a sentry. Annotating it as Any is a nice workaround.

Reply:
Changed in 9763b66986b6c4e966f230594d741296feff500e:
* I created a new string marker since it made more sense to me to use typed markers only when needed
* I also took the opportunity to remove another marker based on a constant string (which could have led to similar issues)

Reply:
> I think a marker should not be typed, it's just a sentry. Annotating it as Any is a nice workaround.

So, before I do another pass, do you prefer that I use `Union[Optional[str], Marker]` or just `# type: ignore`?

Reply:
> So, before I do another pass, do you prefer that I use `Union[Optional[str], Marker]` or just `# type: ignore`?

Well, my bad, it turns out that typing the marker with `Any` is enough to make mypy happy. Somehow I thought I had tried it but I guess I was mistaken...

8a44cc9 should be good to go *crossing fingers*